front-end/src/components/analysis-page/IntentMenu.jsx

```diff
@@ -0,0 +1,165 @@
+                <li className="relative" key={filteredIndex}>
+                    <button
+                        className="w-full text-left text-lg py-0 px-4 h-10 rounded truncate overflow-hidden box-border 
+                        hover:bg-gray-100 hover:text-black transition ease-in-out
+                        focus-within:bg-gray-100 focus-within:text-black"
+                        onClick={goToIntent}
+                        aria-label={intent.question}
+                        data-testid="intent-menu-list-item"
+                    >
+                        <div /* Format the text representation of the intent in the intent menu */>
+                            {this.formatSearchResult(intent.question, searchSlices)}
+                        </div>
+                    </button>
+                </li>
+            );
+        });
+    }
+
+	render() {
+		return (
+            <div /* Ternary operator is used in class below to animate sliding in/out of intent menu depending on isOpen prop */
+                className={"z-10 w-1/4 h-full bg-off-white font-cabin absolute shadow-lg shadow-blue/30 transition ease-in-out " + (this.props.isOpen ? "translate-x-0" : "-translate-x-full")}
+                data-testid="intent-menu"
+                aria-hidden={!this.props.isOpen}
+                ref={this.ref}
+            >
+                {this.props.isOpen && /* Only render internal elements if menu isOpen */
+                    <div className="h-5/6">
+                        <div className="ml-4 mt-4">
+                            <BaseButton
+                                click={this.props.onClickOutside}
+                                label="Close intent menu"
+                                icon={{
+                                    name: "x",
+                                    size: "20",
+                                    color: "black"
+                                }}
+                            />
+                        </div>
+                        <ul className="text-center font-bold text-xl mt-10"> 
+                            Transcript Intents:
+                        </ul>
+                        <ul className="h-4/5 relative overflow-y-scroll bg-off-white-100 shadow-inner-lg rounded p-2 my-4 mx-4">
+                            {this.listIntents()}
+                        </ul>
+                        <ul className="flex flex-col" >
+                            <input /* Transcript search input */
+                                className="bg-off-white text-xl ml-4 mr-4 mt-6 rounded-md px-4 py-2 drop-shadow-md outline-none transition ease-in-out
+                                border border-solid border-purple-100
+                                hover:border-purple-200
+                                focus:border-purple-300 focus:ring-purple-300"
+                                aria-label="Search by intent contents"
```

Comment:
```suggestion
                                aria-label="Search by keyword"
```
I think this will make it less confusing for users.

Reply:
Good call